hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/balancer/MoveManager.java

```diff
@@ -411,17 +404,18 @@ private void sendReplicateCommand(
    *
    * @param containerInfo Container to be deleted
    * @param datanode The datanode on which the replica should be deleted
+   * @param scheduledTime The time at which the replicate command for the container was scheduled
    */
   private void sendDeleteCommand(
-      final ContainerInfo containerInfo, final DatanodeDetails datanode)
+      final ContainerInfo containerInfo, final DatanodeDetails datanode,
+      long scheduledTime)
       throws ContainerReplicaNotFoundException, ContainerNotFoundException,
       NotLeaderException {
     int replicaIndex = getContainerReplicaIndex(
         containerInfo.containerID(), datanode);
     long deleteTimeout = moveTimeout - replicationTimeout;
-    long now = clock.millis();
     replicationManager.sendDeleteCommand(
-        containerInfo, replicaIndex, datanode, true, now + deleteTimeout);
+        containerInfo, replicaIndex, datanode, true, scheduledTime + deleteTimeout);
```

Comment:
The deadline should be `moveStartTime + moveTimeout`, we don't need `deleteTimeout`.